src/movies.js

```diff
@@ -1,25 +1,128 @@
+};
+
+//console.log(getAllDirectors(movies))
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+
+    const total = moviesArray.filter((name) => name.director === 'Steven Spielberg' && name.genre.includes('Drama'));
+      
+    return total.length;
+};
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+
+    if(moviesArray.length === 0) return 0;
+    
+    const totalscore = moviesArray.reduce((acc, val) => {
+        if(!val.score) {
+            return acc;
+        } else {
+            return acc + val.score;
+        }
+    },0);
+
+    averageScore = totalscore / moviesArray.length;
+    
+    //return averageScore.toFixed(2);
+    return Math.round(averageScore * 100) / 100;
+};
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+const movie = [
+    {
+        title: 'The Shawshank Redemption',
+        year: 1994,
+        director: 'Frank Darabont',
+        duration: '2h 22min',
+        genre: ['Crime', 'Drama'],
+        score: 9.3
+      },
+      {
+        title: 'The Godfather',
+        year: 1972,
+        director: 'Francis Ford Coppola',
+        duration: '2h 55min',
+        genre: ['Crime']
+      },
+      {
+        title: 'Dead Poets Society',
+        year: 1989,
+        director: 'Peter Weir',
+        duration: '2h 9min',
+        genre: ['Comedy', 'Drama'],
+        score: 8
+      }
+    ];
+
+function dramaMoviesScore(moviesArray) {
+    if(moviesArray.length === 0) return 0;
+
+    const type = moviesArray.filter((movie) => {
+        return movie.genre.includes('Drama');
+    }); 
+    
+    return scoresAverage(type);
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+    let sortedArray = [...moviesArray];
+    
+    //sortedArray.sort((a, b) => a.year - b.year)
 
-// Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically(moviesArray) {}
+    sortedArray.sort((a, b) => {  //by alphabetical order
+        if(a.year < b.year) return -1;
+        if(a.year > b.year) return 1;
+        if(a.title < b.title) return -1;
+        if(a.title > b.title) return 1;   
+    });
 
+    return sortedArray;
+};
+
+// Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(moviesArray) {
+    let arr = [...moviesArray];
+    
+    const orderMovies = arr.map((name) => name.title);
+    return orderMovies.sort().splice(0, 20);
+  }
+  
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
-function turnHoursToMinutes(moviesArray) {}
+function turnHoursToMinutes(moviesArray) {
```

Comment:
The logic is more or less what you've done here but you were missing some steps:

```
function turnHoursToMinutes(moviesArray) {

  function convertDuration(movie) {
    let total = movie.duration.split(" ");
    let hour = 0
    let min = 0
    if (total[0].includes("h")) {
      hour = total[0].split("h");
      hour = Number(hour[0]) * 60;
    }
    if (total[1] && total[1].includes("min")) {
      min = total[1].split("min");
      min = Number(min[0]);
    }
    total = min + hour;
    return Number(total);
  }

  let newArray = moviesArray.map((oneMovie) => {
    let newMovie = {};
    newMovie.title = oneMovie.title;
    newMovie.year = oneMovie.year;
    newMovie.director = oneMovie.director;
    newMovie.duration = convertDuration(oneMovie);
    newMovie.genre = oneMovie.genre;
    newMovie.score = oneMovie.score;
    return newMovie;
  });
  return newArray;
}
```
First you create a function (convertDuration) inside this function (turnHoursToMinutes) that will convert the duration of only one movie into minutes. 
Then you need to map over the given array and make a copy of every single element of the array but, instead of copying the duration you call the previous function that will convert it into minutes.